### PR TITLE
fix: ust position not reflecting on trading account balance table

### PR DESCRIPTION
**Pull request by ThomasRalee**
## This PR:
- resolves UST position not reflecting on trading account balance table issue

### CODE REVIEW

File: components/partials/portfolio/balances.vue [hunk 1]
```diff
@@ -51,7 +51,7 @@
             <span v-else-if="hideBalance">{{ HIDDEN_BALANCE_DISPLAY }}</span>
             <span v-else>
               {{ $t('common.available') }}
-              {{ totalSubaccountBalancesToString }} USD
+              {{ totalSubaccountAvailableBalancesToString }} USD
```
Comment (bangjelkoski) [inline components/partials/portfolio/balances.vue:54]: Nice variable name change!

File: components/partials/portfolio/balances.vue [hunk 2]
```diff
@@ -273,15 +290,23 @@ export default Vue.extend({
...
     },
 
-    totalPositionsPnl(): BigNumberInBase {
+    totalPositionsPnlByQuoteToken(): Record<string, BigNumberInBase> {
       const { markets, orderbooks, positions } = this
 
-      return positions.reduce((total, p) => {
+      const list = {} as Record<string, BigNumberInBase>
+
+      positions.forEach((p) => {
```
Comment (bangjelkoski) [inline components/partials/portfolio/balances.vue:298]: You should use `reduce` here

File: components/partials/portfolio/balances.vue [hunk 3]
```diff
@@ -303,36 +328,78 @@ export default Vue.extend({
...
     },
 
-    totalPositionsMargin(): BigNumberInBase {
+    totalPositionsMarginByQuoteToken(): Record<string, BigNumberInBase> {
       const { markets, positions } = this
 
-      return positions.reduce((total, p) => {
+      const list = {} as Record<string, BigNumberInBase>
+
+      positions.forEach((p) => {
```
Comment (bangjelkoski) [inline components/partials/portfolio/balances.vue:363]: Same here, `reduce` should be used

**Review by bangjelkoski — CHANGES_REQUESTED** [review]

**Review by bangjelkoski — CHANGES_REQUESTED** [review]

File: components/partials/portfolio/balances.vue [hunk 4]
```diff
@@ -329,9 +325,9 @@ export default Vue.extend({
               .times(p.direction === TradeDirection.Long ? 1 : -1)
 
         list[quoteToken] = list[quoteToken].plus(pnl)
```
[the remark this answers is not in the document]
Comment (ThomasRalee) [reply components/partials/portfolio/balances.vue:327]: The check was added above, it sets the key if it doesn't exist to `ZERO_IN_BASE`
````
if (!list[quoteToken]) {
    list[quoteToken] = ZERO_IN_BASE
}
````

**Review by bangjelkoski — APPROVED** [review]

File: components/partials/portfolio/balances.vue [hunk 5]
```diff
@@ -299,10 +299,10 @@ export default Vue.extend({
           return list
         }
 
-        const quoteToken = market.oracleQuote.toLowerCase()
+        const quoteDenom = market.oracleQuote.toLowerCase()
```
Comment (bangjelkoski) [inline components/partials/portfolio/balances.vue:302]: I was asking about this: `market.oracleQuote.toLowerCase()`, why are we not using `market.quoteDenom.toLowerCase()`

**Pull request merged**